From 18c014093d89c03dbb401d8dbac8cc24be3ce386 Mon Sep 17 00:00:00 2001
From: Mashopy <eliasgheeraert@gmail.com>
Date: Mon, 12 Oct 2026 19:19:10 +0000
Subject: [PATCH] libfs_avb: Allow LKs patched with fenrir to boot on AOSP

fenrir spoof bootloader state to green, however in doing so it result
in AVB failures.

Rebased onto lineage-24.0, where libfs_avb lives in system/fs/fs_mgr.
Original: Nothing-2A/android_system_core@8ff6e7a

Change-Id: I75b28942684c736492ee2a605f3f137dd0d89b2e
---
 libfs_avb/util.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libfs_avb/util.cpp b/libfs_avb/util.cpp
index 79c7b94..f78e89c 100644
--- a/libfs_avb/util.cpp
+++ b/libfs_avb/util.cpp
@@ -111,7 +111,7 @@ bool IsDeviceUnlocked() {
     std::string verified_boot_state;
 
     if (fs_mgr_get_boot_config("verifiedbootstate", &verified_boot_state)) {
-        return verified_boot_state == "orange";
+        return (verified_boot_state == "orange" || verified_boot_state == "green");
     }
     return false;
 }
-- 
2.43.0

